pipert/contrib/video_writer.py

```diff
@@ -0,0 +1,77 @@
+import cv2
+from pipert import BaseComponent, Routine
+from queue import Queue
+import argparse
+from urllib.parse import urlparse
+from imutils import resize
+from pipert.core.mini_logics import MessageFromRedis
+from pipert.core import QueueHandler, Message
+
+
+class VideoWriterLogic(Routine):
+
+    def __init__(self, output_file, in_queue, fps=30, im_size=(640, 480), *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.writer = None
+        self.output_file = output_file
+        self.q_handler = QueueHandler(in_queue)
+        self.w, self.h = im_size
+        self.fps = fps
+
+    def main_logic(self, *args, **kwargs):
+        msg: Message = self.q_handler.non_blocking_get()
+        if msg:
+            frame = msg.get_payload()
+            if frame.shape[0] != self.h or frame.shape[1] != self.w:
+                self.writer.release()
+                self.writer = cv2.VideoWriter(self.output_file, cv2.VideoWriter_fourcc('M', 'J', 'P', 'G'), self.fps,
+                                              (frame.shape[1], frame.shape[0]))
+                self.h, self.w = frame.shape[0], frame.shape[1]
+
+            frame = cv2.putText(frame, f"{msg.id.split('_')[-1]}", (0, self.h - 10), cv2.FONT_HERSHEY_SIMPLEX,
+                                1, (0, 0, 255), 2, cv2.LINE_AA)
+            print(frame.shape)
```

Comment:
Is it necessary?

Reply:
Of course not, part of the debugging I missed, good catch :)